agent/agent.go

```diff
@@ -832,7 +832,7 @@ func (a *Agent) listenAndServeDNS() error {
 			merr = multierror.Append(merr, err)
 		case <-timeout:
 			merr = multierror.Append(merr, fmt.Errorf("agent: timeout starting DNS servers"))
-			break
+			return merr.ErrorOrNil()
```

Comment:
The `break` was doing nothing because it was only breaking out of the select/case, and there is nothing after the select in the loop.

I assume this meant to break out of the for loop, since this is a timeout, so changing to `return` seems like the right fix.